Cython/Compiler/MemoryView.py

```diff
@@ -835,26 +843,43 @@ def use_cython_array_utility_code(env):
 
 is_contig_utility = load_memview_c_utility("MemviewSliceIsContig", context)
 overlapping_utility = load_memview_c_utility("OverlappingSlices", context)
-copy_contents_new_utility = load_memview_c_utility(
-    "MemviewSliceCopyTemplate",
-    context,
-    requires=[],  # require cython_array_utility_code
-)
-
-view_utility_code = load_memview_cy_utility(
-        "View.MemoryView",
-        context=context,
-        requires=[Buffer.buffer_struct_declare_code,
-                  Buffer.buffer_formats_declare_code,
-                  memviewslice_init_code,
-                  is_contig_utility,
-                  overlapping_utility,
-                  copy_contents_new_utility,
-                  ],
-)
+
+
+def _get_copy_contents_new_utility():
+    copy_contents_new_utility = load_memview_c_utility(
+        "MemviewSliceCopyTemplate",
+        context,
+        requires=[],  # require cython_array_utility_code
+    )
+    return copy_contents_new_utility
+
+@Utils.cached_function
+def _get_memoryview_utility_code():
```

Comment:
So, you created all these factory functions that replace the utility code constants with (apparently) fresh, constant copies of the utility code. And now you cache the creation to return always the same objects? What am I missing here?

Reply:
It's targeted at improving #6531.

It lets you generate memoryview code based on env (i.e. compiler directives rather than `Options`). So it's specifically aimed at making `shared_utility_code` as a directive.

But you're right that it achieves nothing on its own.